### PR TITLE
parse dependency specifier correctly

### PR DESCRIPTION
**Pull request by alexlamsl**

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
this definitely looks better, but you should try to run the tests at the bottom of semver.zig

**Comment by alexlamsl** [comment]
```console
$ zig test src/install/semver.zig
src/install/semver.zig:1:21: error: no package named 'bun' available within package 'root'
const bun = @import("bun");
                    ^~~~~
referenced by:                                                                                                                                              
    string: src/install/semver.zig:2:16
    test.String works: src/install/semver.zig:417:18
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```
... or do you mean translating those tests into `bun wiptest` format?

File: src/install/dependency.zig [hunk 1]
```diff
@@ -303,147 +303,148 @@ pub const Version = struct {
...
-                '.' => return Tag.folder,
-
+                // =1
+                // >1.2
+                // >=1.2.3
+                // <1
+                // <=1.2
+                // ^1.2.3
+                // *
+                '=', '>', '<', '^', '*' => return .npm,
```
Comment (Jarred-Sumner) [inline src/install/dependency.zig]: This needs to be `|` in the case where someone does something like

```zig
|| 6.0.0
```

Comment (Jarred-Sumner) [reply src/install/dependency.zig]: which believe it or not i think i saw once

Comment (alexlamsl) [reply src/install/dependency.zig]: good thing it's undocumented :smirk:

Comment (alexlamsl) [reply src/install/dependency.zig]: Done in https://github.com/oven-sh/bun/pull/1840/commits/fa5b80869b58401e7850cec15feb7e3c4ea2d2c8

**Pull request merged**